src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -236,64 +300,79 @@ object DeployPlugin extends AutoPlugin {
       log.info("Press ENTER to continue with no environment configuration, CTRL-C to abort.")
       log.info("")
       System.console.readLine()
+      ConfigFactory.empty
     }
 
-    val deploys: Seq[Future[Try[Unit]]] = deployConfig.hostConfigs map { hostConfig =>
-      Future {
-        // Command to pass to rsync's "rsh" flag, and to use as the base of our ssh operations.
-        val sshCommand = {
-          val keyFile = sys.env.getOrElse("AWS_PEM_FILE", {
-            throw new IllegalStateException(s"Environment variable 'AWS_PEM_FILE' is undefined")
-          })
-          require(keyFile.nonEmpty, "Environment variable 'AWS_PEM_FILE' is empty")
-
-          Seq("ssh", "-i", keyFile, "-l", hostConfig.sshUser)
-        }
-        val deployHost = hostConfig.host
-        val deployDirectory = hostConfig.directory
-
-        val rsyncDirs = deployDirs.value map (name => s"--include=/$name")
-        val rsyncCommand = Seq.concat(
-          Seq("rsync", "-vcrtzP", "--rsh=" + sshCommand.mkString(" ")),
-          rsyncDirs,
-          Seq(
-            "--exclude=/*",
-            "--delete",
-            universalStagingDir.getPath + "/",
-            deployHost + ":" + deployDirectory
-          )
-        )
-
-        // Shell-friendly version of rsync command, with rsh value quoted.
-        val quotedRsync = rsyncCommand.patch(
-          2, Seq("--rsh=" + sshCommand.mkString("\"", " ", "\"")), 1
-        ).mkString(" ")
-
-        // Now, ssh to the remote host and run the restart script.
-        val restartScript = deployDirectory + "/" + hostConfig.startupScript
-        val restartArgs = if (hostConfig.startupArgs.nonEmpty) {
-          "--" +: hostConfig.startupArgs
-        } else {
-          Seq()
-        }
+    // Command to pass to rsync's "rsh" flag, and to use as the base of our ssh operations.
+    val sshCommand = {
+      val keyFile = sys.env.getOrElse("AWS_PEM_FILE", {
```

Comment:
Now that my brain's working again, I think this can be made optional - and we just omit the `-i` flag from `ssh` if it's missing. Most users should have passwordless ssh set up for the prod deploy user. Maybe add a TODO note about this?